### PR TITLE
Pv2 1325 submission event service

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/SubmissionEventProcessorTest.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            Assert.AreEqual(event2.FamilyName, updatedSubmissionChanged.IlrDetails.FamilyName);
+            Assert.AreEqual(event2.CompStatus, updatedSubmissionChanged.IlrDetails.CompStatus);
+            // these 3 should always be supplied, even if they haven't changed (although their change triggers an event)...
+            Assert.AreEqual(event2.CommitmentId, updatedSubmissionChanged.IlrDetails.CommitmentId);
+            Assert.AreEqual(event2.EPAOrgId, updatedSubmissionChanged.IlrDetails.EPAOrgId);
+            Assert.AreEqual(event2.EmployerReferenceNumber, updatedSubmissionChanged.IlrDetails.EmployerReferenceNumber);
+        }
+
+        [Test]
+        public void UpdatesLastSeenVersionToBeCurrentVersion()
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/SubmissionEventProcessorTest.cs]: I can't see any difference between this and the first method except the ExmploterReferenceNumber. Is it doing anything different?

File: src/SFA.DAS.Payments.EarningEvents.Model/Entities/SubmittedPriceEpisodeEntity.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public class SubmittedPriceEpisodeEntity
+    {
+        public long Id { get; set; }
+        public long Ukprn { get; set; } // ilr, have
+        public string LearnRefNumber { get; set; } // aim, have
+        public string PriceEpisodeIdentifier { get; set; } // multiple per aim, have
+        public IlrDetails IlrDetails { get; set; }
+    }
+
+    public class IlrDetails
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.EarningEvents.Model/Entities/SubmittedPriceEpisodeEntity.cs]: should be in its own file

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/SFA.DAS.Payments.AcceptanceTests.EndToEnd.csproj [hunk 3]
```diff
@@ -194,7 +194,7 @@
       <HintPath>..\packages\NServiceBus.Autofac.7.0.0\lib\net452\NServiceBus.ObjectBuilder.Autofac.dll</HintPath>
     </Reference>
     <Reference Include="NServiceBus.Persistence.AzureStorage, Version=2.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
-      <HintPath>..\packages\NServiceBus.Persistence.AzureStorage.2.3.0\lib\net452\NServiceBus.Persistence.AzureStorage.dll</HintPath>
+      <HintPath>..\packages\NServiceBus.Persistence.AzureStorage.2.2.0\lib\net452\NServiceBus.Persistence.AzureStorage.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/SFA.DAS.Payments.AcceptanceTests.EndToEnd.csproj]: looks fine - the rest of the solution uses 2.2

File: src/SFA.DAS.Payments.EarningEvents.Application/Interfaces/SubmissionEventGeneratorService.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    PriceEpisodeIdentifier = e.Identifier,
+                    IlrDetails = new IlrDetails
+                    {
+                        IlrFileName = earningEvent.IlrFileName,
+                        AcademicYear = earningEvent.CollectionPeriod.AcademicYear.ToString(),
+                        ActualStartDate = e.CourseStartDate,
+                        PlannedEndDate = e.PlannedEndDate,
+                        ActualEndDate = e.ActualEndDate,
+                        AimSeqNumber = earningEvent.LearningAim.SequenceNumber,
+                        //CommitmentId =
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.EarningEvents.Application/Interfaces/SubmissionEventGeneratorService.cs]: removed

File: src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/SubmissionEventProcessorTest.cs (hunk 1 above)
Comment (davehowlett) [reply src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/SubmissionEventProcessorTest.cs]: removed as the last test didn't test anything additional

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/packages.config [hunk 5]
```diff
@@ -45,7 +45,7 @@
   <package id="NServiceBus.Azure.Transports.WindowsAzureServiceBus" version="9.0.3" targetFramework="net471" />
   <package id="NServiceBus.Callbacks" version="3.0.0" targetFramework="net471" />
   <package id="NServiceBus.Newtonsoft.Json" version="2.1.0" targetFramework="net471" />
-  <package id="NServiceBus.Persistence.AzureStorage" version="2.3.0" targetFramework="net471" />
+  <package id="NServiceBus.Persistence.AzureStorage" version="2.2.0" targetFramework="net471" />
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/packages.config]: all other projects 2.2. This is fine

File: src/SFA.DAS.Payments.EarningEvents.Application/Repositories/SubmittedPriceEpisodeRepository.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly IPaymentsDataContext paymentsDataContext;
+        private readonly IPaymentLogger logger;
+
+        public SubmittedPriceEpisodeRepository(IPaymentsDataContext paymentsDataContext, IPaymentLogger logger)
+        {
+            this.paymentsDataContext = paymentsDataContext;
+            this.logger = logger;
+        }
+
+        public async Task<List<SubmittedPriceEpisodeEntity>> GetLastSubmittedPriceEpisodes(long ukprn, string learnRefNumber, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.EarningEvents.Application/Repositories/SubmittedPriceEpisodeRepository.cs]: Removed the "Last" from method name

File: src/SFA.DAS.Payments.EarningEvents.Model/Entities/SubmittedPriceEpisodeEntity.cs (hunk 2 above)
Comment (davehowlett) [reply src/SFA.DAS.Payments.EarningEvents.Model/Entities/SubmittedPriceEpisodeEntity.cs]: moved file

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/packages.config (hunk 5 above)
[the remark this answers is not in the document]
Comment (davehowlett) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/packages.config]: the rest of the solution is 2.2 so this is fine.

**Comment by MichaelYoung1981** [comment]
Moved back to analysis.

**Pull request closed without merging**